Makefile

```diff
@@ -24,25 +31,61 @@ NVCCFLAGS= -std=c++14 -ccbin=$(CXX) -O3 $(NVARCH) -Wno-deprecated-gpu-targets \
 # and enable cufinufft internal flags.
 #NVCCFLAGS+= -DINFO -DDEBUG -DRESULT -DTIME
 
-# CUDA Related build dependencies
-CUDA_ROOT=/usr/local/cuda
-INC=-I$(CUDA_ROOT)/include \
-	-Icontrib/cuda_samples
-NVCC_LIBS_PATH=-L$(CUDA_ROOT)/lib64
+# CUDA Related build dependencies -- the user can overwrite CUDA_ROOT using the
+# CUDA_DIR environment variable. If neither (CUDA_ROOT, nor CUDA_DIR) is set,
+# CUDA_ROOT defaults to /usr/local/cuda
+ifeq ($(CUDA_DIR),)
+    CUDA_ROOT ?= /usr/local/cuda
+else
+    CUDA_ROOT := $(CUDA_DIR)
+endif
+
+# Common includes
+INC=-I$(CUDA_ROOT)/include -Icontrib/cuda_samples
+ifdef FFTW_INC
+    $(info detected a FFTW_INC variable -- setting FFTW include directory)
+    INC += -I$(FFTW_INC)
+endif
 
 FFTWNAME=fftw3
 
+# Common libs
 LIBS=-lm -lcudart -lstdc++ -lnvToolsExt -lcufft -lcuda -l$(FFTWNAME) -l$(FFTWNAME)f
+ifdef FFTW_DIR
+    $(info detected a FFTW_DIR variable -- setting FFTW library directory)
+    LIBS += -L$(FFTW_DIR)
+endif
+
+# NVCC-specific includes
+NVCC_INC=
+ifdef FFTW_INC
+    NVCC_INC += -I$(FFTW_INC)
+endif
+
+# NVCC-specific libs
+NVCC_LIBS_PATH = -L$(CUDA_ROOT)/lib64
+ifdef FFTW_DIR
+    NVCC_LIBS_PATH += -L$(FFTW_DIR)
+endif
+ifdef NVCC_STUBS
+    $(info detected CUDA_STUBS -- setting CUDA stubs directory)
+    NVCC_LIBS_PATH += -L$(NVCC_STUBS)
+endif
+
 
 
 #############################################################
 # Allow the user to override any variable above this point. #
--include make.inc
+uname_p := $(shell uname -p)
+ifeq ($(uname_p), ppc64le)
+    -include make.inc.power9
```

Comment:
If a power9 user has a `make.inc` (I would if I was on Traverse at Princeton for example), this may override it.

We might want to always take the `make.inc` if it is available. It doesn't ship with our code, so a user would have to intentionally create or symlink it...

With that said, taking the best default otherwise makes sense to me, in the case `make.inc` dne.  

@ahbarnett , do you have a preference here so we stay inline with FINUFFT? Thanks

Reply:
I see where you're going with this: `make.inc` => local user settings. I'm happy with this approach. The design here would then be:

1. `make.inc` -> local user settings
2. `make.inc.arch` -> architecture-specific settings
3. `site/**` -> site (like compute center, etc) specific settings

Reply:
Yes that would be okay by me, but we would want to stay consistent with our related package FINUFFT, so the notion of a `site` directory for make include files should get run by both projects (I tagged the projects' lead for you).  It may be suggested to keep things "flat" for simplicity.  I'm just helping out with the project recently; I'm sure FI team will give you some feedback here (they are eastern time so maybe tomorrow.)

Reply:
Thanks. I'll keep an eye out for what they stay!

Reply:
Hello Johannes & Garrett,

I am fine you adding a site directory of more make.inc.* files, as long as you document it.
In our CI for FINUFFT we include eg
`cp make.inc.macosx_clang make.inc`
as part of the docker setup.
So the set of make.inc.* will become more reliable and system-specific.

I don't think the main makefile should have the OS-specific checks though - will become too messy. Maybe a separate script or makefile which sets up make.inc for the system/site is best? I'm not rigid about this though.

Reply:
Hey @ahbarnett 

I've gone and removed the architecture check for power9 -- technically `uname_p := $(shell uname -p)` checks for the cpu architecture, so I wouldn't call it OS-specific checks (you're already running make 😉 ). But I know what you mean. 

Have a look at the beginning of `Makefile` now -- I've implemented a compromise solution, where the user can specify a site and machine architecture using eg `make site=<computer> target=<archtiecture>`. The sites can also override architectures.

In keeping with the design, I've also moved the targets into a dedicated `targets` directory -- and `make.inc` can be used to overwrite variables locally (as before).

Let me know what you think.